### ISSUE TITLE
ValueError: not enough values to unpack (expected 2, got 1)

### ISSUE THREAD

**Issue by bblanchon** [jazzband/django-debug-toolbar#1612]
Hi,

Thank you very much for this essential library ❤️ 

I just upgraded my project to Django 4.0.4 and now I frequently get the following exception when refreshing the page:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.9/site-packages/debug_toolbar/middleware.py", line 67, in __call__
    panel.generate_stats(request, response)
  File "/usr/local/lib/python3.9/site-packages/debug_toolbar/panels/sql/panel.py", line 230, in generate_stats
    query["stacktrace"] = render_stacktrace(query["stacktrace"])
  File "/usr/local/lib/python3.9/site-packages/debug_toolbar/utils.py", line 75, in render_stacktrace
    directory, filename = abspath.rsplit(os.path.sep, 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

The value of `abspath` is `'<frozen importlib._bootstrap>'`.

Let me know if you need further information.

Best regards,
Benoit

**Comment by knyghty** [comment]
I'm getting the same kind of errors during tests after upgrading to 3.3.0, so the problem was likely introduced in this release.

**Comment by bblanchon** [comment]
Indeed, I'm using django-debug-toolbar 3.3.0.

**Comment by matthiask** [comment]
Thanks for the report! I'm unable to reproduce this (I tried with Python 3.9 and 3.10 in Linux/WSL)

Do you have a way to print the stack which fails, e.g. by printing the value of `query["stacktrace"]`? That may prove useful.

**Comment by knyghty** [comment]
It's a little hard to reproduce on my end. It only happens during tests (that I've noticed so far, anyway) and only when running pytest with xdist - so the tests running in parallel - and you can't use --pdb when running in parallel. Maybe that information is useful somehow.

**Comment by bblanchon** [comment]
The exception occurs every time I hit refresh after a status code 500.

To reproduce:

1. Edit a view to insert a bug (for example, an undefined symbol).
2. Open the corresponding URL in your browser.
3. Wait for the Django error page to show.
4. Remove the bug from the view
5. Refresh the page in the browser
6. No error is expected, yet you can see "not enough values to unpack (expected 2, got 1)"

**Comment by tim-schilling** [comment]
So what I think is happening is that `render_stacktrace` is being passed a list of frame data and expects the first frame to be the path, but it's actually the frame.

**Comment by tim-schilling** [comment]
@matthiask I don't think we're going to figure out what path and file to show when abspath is set to `<frozen importlib._bootstrap>`. This was probably always an issue but hidden by the Django template rendering system that was removed in #1571.

I think a reasonable solution would be:

```python
if os.path.sep in abspath:
    directory, filename = abspath.rsplit(os.path.sep, 1)
else:
    directory = ""
    filename = abspath
```

I'm working on the PR to add the test to prevent regressions.

**Comment by tim-schilling** [comment]
@bblanchon @knyghty could one of you confirm that #1613 fixes the issue for you locally?

**Comment by knyghty** [comment]
@tim-schilling yes, my tests are passing now with that patch at least.

**Comment by bblanchon** [comment]
```python
code = ''
func = '_find_andload'
lineno = 1007
locals_ = None
```

#1613 does fix the issue 👍

**Comment by tim-schilling** [comment]
This fix was shipped in version 3.4 today.
